tasks/averages/moving-average/src/main.rs

```diff
@@ -0,0 +1,66 @@
+struct MovingAverage {
+    period: u32,
+    list: Vec<f32>,
+}
+
+#[derive(Debug)]
+struct MovingAverageError {
+    message: String,
+}
+
+impl MovingAverage {
+    fn new(period: u32) -> MovingAverage {
+        if period > 0 {
```

Comment:
Note that the error talks about period being lower than 0, but it will also be triggered if period is equal to 0 (which is the only way to trigger the error, as @Hoverbear points out)